ユーザー管理機能の実装

matsu006007 opened the pull request:
# what
ユーザー管理機能の実装

# why
ユーザーを管理できるようにするため

matsu006007 committed:
実装①

matsu006007 committed:
エラー確認　修正

matsu006007 committed:
コメント削除、ケアレスミス修正

matsu006007 committed:
Rspecでのモデル単体テストコード入力

matsu006007 commented:
申し訳ありません。
Rubocopを実行するのを忘れてしまいました。

matsu006007 commented:
Gyazo GIF URL（ログイン→ログアウト）
https://gyazo.com/177cf507b5a4da2d05ba63814a34ed9f

Gyazo URL（テストコードの結果）
https://gyazo.com/d3f91256031432240185b488d6b72043

maimai0401 commented:
レビュー依頼ありがとうございます！
こちらのプルリクエストですが、実際の画面や機能ができているか確認させていただきたいので、下記キャプチャ動画を追加でご提出いただけますか？

・新規登録に失敗し、エラーが表示される動画
・ログインに成功する動画
・ログインに失敗し、エラーが表示される動画

「Gyazo　使い方」とgoogleで検索し、ツールによって生成されたURLをこのPRに貼っていただければ幸いです。
その後、再度レビュー依頼フォームにてご提出をお願いいたします。

matsu006007 commented:
Gyazo GIF URL

・新規登録に失敗し、エラーが表示される動画
　　https://gyazo.com/488a556e5bf2b576644cd2563a8201c1

・ログインに成功する動画
　　https://gyazo.com/c22818f12ef66e227e2574c2a6a6deb4

・ログインに失敗し、エラーが表示される動画
　　https://gyazo.com/b05fcba58e930271039588bfdb2e3d08

kazuki0122 commented on app/models/user.rb:
```diff
@@ -0,0 +1,26 @@
+  with_options presence: true, format: { with: /\A[ァ-ヶ]+\z/} do
+    validates :last_name_kana
+    validates :first_name_kana
+  end
```
presence: trueを使用してる箇所は全てwith_optionsを使って共通化することができるので修正しましょう！以下が使用例です！ 
```
with_options presence: true do
    validates :nickname
　省略　
end
```

kazuki0122 commented on spec/models/user_spec.rb:
```diff
@@ -0,0 +1,138 @@
+      it "ユーザー登録情報と本人情報が全て存在すれば登録できる" do
+        expect(@user).to be_valid
+      end
+      it "passwordとpassword_confirmationが６文字以上であれば登録できる" do
+        @user.password = "123abc"
+        @user.password_confirmation = "123abc"
+        expect(@user).to be_valid
+      end
+      it "last_nameとfirst_nameが漢字であれば登録できる" do
+        @user.last_name = "漢字"
+        @user.first_name = "漢字"
+        expect(@user).to be_valid
+      end
+      it "last_nameとfirst_nameが全角カタカナであれば登録できる" do
+        @user.last_name = "カタカナ"
+        @user.first_name = "カタカナ"
+        expect(@user).to be_valid
+      end
+      it "last_nameとfirst_nameが全角ひらがなであれば登録できる" do
+        @user.last_name = "ひらがな"
+        @user.first_name = "ひらがな"
+        expect(@user).to be_valid
+      end
```
こちらのテストですが、```it "ユーザー登録情報と本人情報が全て存在すれば登録できる" do```で行ってるテストと重複するので今回は不要となります！削除しましょう！

kazuki0122 reviewed (commented):
レビュー依頼ありがとうございます！
コメント致しましたので、ご確認お願い致します！

matsu006007 committed:
with_options　記述

matsu006007 committed:
user_spec.rb  テストコード削除

①passwordとpassword_confirmationが６文字以上であれば登録できる

②last_nameとfirst_nameが漢字であれば登録できる

③last_nameとfirst_nameが全角カタカナであれば登録できる

④last_nameとfirst_nameが全角ひらがなであれば登録できる

上記４つのテストコードを削除

masayaomoto commented on spec/models/user_spec.rb:
```diff
@@ -0,0 +1,118 @@
+        @user.password_confirmation = 'aaaaaa'
+        @user.valid?
+        expect(@user.errors.full_messages).to include('Password is invalid')
+      end
```
passwordに関しまして以下のテストも追加しましょう。
```
passwordが全角の場合は登録できない
```

masayaomoto reviewed (commented):
レビュー依頼ありがとうございます！
修正点確認させていただきました。
追加で1点コメントいたしましたので、ご確認お願いします！

matsu006007 committed:
テストコード　"passwordが全角の場合は登録できない"  追加

matsu006007 commented:
テスト　”passwordが全角の場合は登録できない” ターミナル確認
Gyazo　URL
https://gyazo.com/2c6f6996b580727e1ff7ea431c539a14

masayaomoto commented on app/models/user.rb:
```diff
@@ -0,0 +1,22 @@
+  with_options presence: true do
+    validates :nickname, presence: true
+    validates :birthday, presence: true
+    validates :password_confirmation, presence: true
```
```
validates :nickname, presence: true
validates :birthday, presence: true
validates :password_confirmation, presence: true
```
こちらですが、```with_options presence: true```のdo~end内に記述していただいておりますので、
10,11,12行目の```presence: true```は削除しましょう。

masayaomoto reviewed (commented):
レビュー依頼ありがとうございます！
修正点確認させていただきました。
追加でもう1点コメントいたしましたので、ご確認お願いします！

matsu006007 committed:
記述の修正

masayaomoto commented:
レビュー依頼ありがとうございます！
修正確認いたしました。
LGTMとさせていただきます！実装お疲れ様でした！
引き続き実装頑張っていきましょう！

matsu006007 commented:
ご確認ありがとうございました。
引き続きよろしくお願い致します。

matsu006007 committed:
RuboCopでの修正　１

matsu006007 committed:
RuboCopでの修正　２

matsu006007 committed:
RuboCopでの修正　３

matsu006007 committed:
RuboCopでの修正　４

matsu006007 committed:
RuboCopでの修正５

matsu006007 committed:
RuboCopでの修正　６

matsu006007 committed:
RuboCopでの修正　７

matsu006007 committed:
RuboCopでの修正　８
